protocols/2905a4/2905a4.ot2.apiv2.py

```diff
@@ -0,0 +1,333 @@
+                                starting_well])]
+
+    # Sample Wells
+    tc_plate_wells = tc_plate.wells()[
+                        starting_well_index:starting_well_index+samples]
+    mag_plate_wells = mag_plate.wells()[
+                        starting_well_index:starting_well_index+samples]
+    temp_plate_wells = temp_plate.wells()[
+                        starting_well_index:starting_well_index+samples]
+
+    # Helper Functions
+    def pick_up(pip, loc=None):
+        """Function that can be used instead of .pick_up_tip() that will pause
+        robot when robot runs out of tips, prompting user to replace tips
+        before resuming"""
+        try:
+            if loc:
+                pip.pick_up_tip(loc)
+            else:
+                pip.pick_up_tip()
+        except protocol_api.labware.OutOfTipsError:
+            pip.home()
+            pip.pause("Please replace the empty tip racks!")
+            pip.reset_tipracks()
+            pip.pick_up_tip()
+
+    def remove_supernatant(vol, src, dest, side, mode=None):
+        if mode == 'elution':
+            p300.flow_rate.aspirate = 10
+        else:
+            p300.flow_rate.aspirate = 20
+        while vol > 200:
+            p300.aspirate(
+                200, src.bottom().move(types.Point(x=side, y=0, z=0.5)))
+            p300.dispense(200, dest)
+            p300.aspirate(10, dest)
+            vol -= 200
+        p300.aspirate(vol, src.bottom().move(types.Point(x=side, y=0, z=0.5)))
+        p300.dispense(vol, dest)
+        if dest == trash:
+            p300.blow_out()
+        p300.flow_rate.aspirate = 50
+
+    def reset_flow_rates():
+        p300.flow_rate.aspirate = 46.43
+        p300.flow_rate.dispense = 46.43
+
+    # PROTOCOL STEPS
+    # Set Thermocycler at 4C
+    tc_mod.set_block_temperature(tc_temp)
+
+    # Dilute Magnetic Beads (1)
+    pick_up(p300, available_tips[0])
+    for dest in mag_plate_wells:
+        p300.aspirate(70, rfw.bottom(70))
+        p300.dispense(70, dest.bottom(2))
+        p300.blow_out()
+    p300.return_tip()
+
+    # 30uL of Magnetic Beads to Purification Plate (2)
+    pick_up(p300, available_tips[0])
+    for dest in mag_plate_wells:
+        p300.flow_rate.dispense = 300
+        p300.aspirate(30, beads.bottom(25))
+        p300.dispense(30, dest.bottom(2))
+        p300.mix(7, 75)
+        p300.touch_tip()
+        p300.blow_out()
+    p300.return_tip()
+    reset_flow_rates()
+
+    # Recover beads (3)
+    mag_mod.engage(height=engage_height)
+    # Attracting beads (4)
+    ctx.delay(minutes=2, msg='Attracting beads')
+
+    # Remove supernatant (5)
+    pick_up(p300, available_tips[0])
+    for well in mag_plate_wells:
+        remove_supernatant(120, well, trash, -1)
+    p300.drop_tip()
+    available_tips.pop(0)
+
+    # Setup Dedicated Tips
+    dedicated_tips = dict(zip(range(samples), available_tips[:samples]))
+    available_tips = available_tips[samples::]
```

Comment:
I like this - maybe you can share at the code review

Reply:
Will do!